src/main/java/org/example/FestivalGate.java

```diff
@@ -0,0 +1,33 @@
+package org.example;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import static org.example.TicketType.*;
+
+/**
+ * Blueprint class for the festival gate, where tickets are stored and listed
+ */
+public class FestivalGate {
+    List<TicketType> ticketsList = new ArrayList<>();
+
+    public void addTicket (TicketType ticket) {
```

Comment:
this should be a syncronised list since you don't want two attendees competing for the same arraylist slot